opentelemetry/src/trace/tracer_provider.rs

```diff
@@ -45,40 +45,11 @@ pub trait TracerProvider {
         )
     }
 
-    /// Returns a new versioned tracer with a given name.
```

Comment:
I wouldn't necessarily remove the old way immediately. We could instead mark it deprecated and encourage use of the new one.

Reply:
I have a different opinion on this. I am okay with removing this right away (_i.e in this PR itself_). When we do the next release (0.23 from 0.22), it's allowed to have breaking changes. And in my opinion, its better to clean up **now** than delaying actual removal for another release/accumulating debt for future release.
I agree its more disruptive for end users, but I prefer to do things sooner than delaying. (users anyway have to change code at some point, why not make them do it earlier.)

@hdost @TommyCpp @lalitb do you strongly feel we should keep old api for another release? If everyone prefer to keep this for now, I am okay to accept that. Maybe I am missing something.

Reply:
I vote for being less disruptive for API-related changes, and follow deprecation followed by removal.

Reply:
I generally are OK if we move fast for API. But in this case I think it's a widely used API so it's probably worth a deprecation + remove in next release to control the size of changes needed for a version bump

Reply:
Thanks for sharing your thoughts. I agree. Lets proceed with "mark-deprecated-first, then actual removal in next release" approach for anything very common, like in this case.

Reply:
If we plan to selectively deprecate API changes, and somtimes also directly have breaking changes in consecutive releases, we can add this explicitly in README, to set a clear expectation. Even with the status (alpha, beta) indication, users can expect the breaking changes to be deprecated first.

Have seen some concern raised by the user in twitter, so probably thinking more in that aspect :)

![image](https://github.com/open-telemetry/opentelemetry-rust/assets/1196320/b4804fa4-faa4-4dff-aa52-696d0cd442cc)

Reply:
Yeah sure. Will add this in the readme for the crates.